pkg/operation/botanist/component/kubecontrollermanager/kube_controller_manager.go

```diff
@@ -290,27 +306,84 @@ func (k *kubeControllerManager) Deploy(ctx context.Context) error {
 		return err
 	}
 
-	if _, err := controllerutil.CreateOrUpdate(ctx, k.seedClient, vpa, func() error {
-		vpa.Spec.TargetRef = &autoscalingv1.CrossVersionObjectReference{
-			APIVersion: appsv1.SchemeGroupVersion.String(),
-			Kind:       "Deployment",
-			Name:       v1beta1constants.DeploymentNameKubeControllerManager,
+	if k.hvpaConfig != nil && k.hvpaConfig.Enabled {
+		if err := kutil.DeleteObject(ctx, k.seedClient, vpa); err != nil {
+			return err
 		}
-		vpa.Spec.UpdatePolicy = &autoscalingv1beta2.PodUpdatePolicy{
-			UpdateMode: &vpaUpdateMode,
-		}
-		vpa.Spec.ResourcePolicy = &autoscalingv1beta2.PodResourcePolicy{
-			ContainerPolicies: []autoscalingv1beta2.ContainerResourcePolicy{{
-				ContainerName: containerName,
-				MinAllowed: corev1.ResourceList{
-					corev1.ResourceCPU:    resource.MustParse("100m"),
-					corev1.ResourceMemory: resource.MustParse("100Mi"),
+
+		var (
+			updateModeAuto = hvpav1alpha1.UpdateModeAuto
+			vpaLabels      = map[string]string{v1beta1constants.LabelRole: "kube-controller-manager-vpa"}
+		)
+
+		if _, err := controllerutil.CreateOrUpdate(ctx, k.seedClient, hvpa, func() error {
+			hvpa.Labels = getLabels()
+			hvpa.Spec.Replicas = pointer.Int32Ptr(1)
+			hvpa.Spec.Hpa = hvpav1alpha1.HpaSpec{
+				Deploy:   false,
+				Selector: &metav1.LabelSelector{MatchLabels: getLabels()},
+				Template: hvpav1alpha1.HpaTemplate{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: getLabels(),
+					},
+					Spec: hvpav1alpha1.HpaTemplateSpec{
+						MinReplicas: pointer.Int32Ptr(int32(1)),
+						MaxReplicas: int32(1),
```

Comment:
For `etcd` we use the current replica count, i.e., it is set to `0` during hibernation. Can you elaborate why this one is treated differently?